packages/opentelemetry-core/src/trace/sampler/ParentBasedSampler.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Attributes, Link, Sampler, SamplingResult, SpanContext, SpanKind, TraceFlags } from '@opentelemetry/api';
+import { AlwaysOffSampler } from './AlwaysOffSampler';
+import { AlwaysOnSampler } from './AlwaysOnSampler';
+
+/**
+ * A composite sampler that either respects the parent span's sampling decision
+ * or delegates to `delegateSampler` for root spans.
+ */
+export class ParentBasedSampler implements Sampler {
+  private root: Sampler;
+  private remoteParentSampled: Sampler;
+  private remoteParentNotSampled: Sampler;
+  private localParentSampled: Sampler;
+  private localParentNotSampled: Sampler;
+
+  constructor(config: ParentBasedSamplerConfig) {
+    this.root = config.root;
```

Comment:
Maybe an early check is needed here to ensure pure JavaScript developers can get errors early if the root sampler is not specified?

Reply:
728bdd6